src/utils/privileges.cpp

```diff
@@ -26,6 +28,21 @@
 namespace my_cap {
 
 #ifdef ANTINET_linux
+static void change_user_if_root() {
+	uid_t uid = getuid();
+	if (uid != 0) return; // not root
+	const char* sudo_user_env = std::getenv("SUDO_USER");
+	if (sudo_user_env == nullptr) throw std::runtime_error("Cannot read SUDO_USER env") ; // get env error
```

Comment:
this can happen when root runs the program directly.
In this case, we should throw custom exception no_regular_user_to_exit_sudo("SUDO_USER is not set")